### PR TITLE
Ops 1232 agreement filter button

### PR DESCRIPTION
**Pull request by johndeange**
## What changed

Added HTML/CSS for filter button.

N.B. The filter functionality has not been added yet, i.e. only the visual appearance for now.  That work with be next.

## Issue

https://github.com/HHS/OPRE-OPS/issues/1232

## How to test

Verify the visual appearance of the filter modal matches the design.

## Screenshots

image

### CODE REVIEW

File: frontend/src/pages/agreements/list/AgreementsFilterButton.module.css [hunk 1]
```diff
@@ -0,0 +1,33 @@
+.filterButton {
+  background-color: #162E51;
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/list/AgreementsFilterButton.module.css:2]: wonder if we have global colors we can use?

File: frontend/src/pages/agreements/list/AgreementsFilterHeaderSection.jsx [hunk 2]
```diff
@@ -39,9 +40,18 @@ export const AgreementsFilterHeaderSection = () => {
         );
     });
 
+    // useEffect(() => {
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/list/AgreementsFilterHeaderSection.jsx:43]: imagine this was left in for next step?

File: frontend/src/pages/agreements/list/AgreementsFilterButton.module.css (hunk 1 above)
Comment (johndeange) [reply frontend/src/pages/agreements/list/AgreementsFilterButton.module.css:2]: I looked for it and didn't see it used elsewhere in the app.  In USWDS it isn't used other than for links where in this case it wouldn't semantically make sense.

**Comment by johndeange** [comment]
@fpigeonjr I will look for a way to close the modal when you click outside of it - not sure if it's supported with the `react-modal` library

File: frontend/src/pages/agreements/list/AgreementsFilterButton.module.css (hunk 1 above)
Comment (fpigeonjr) [reply frontend/src/pages/agreements/list/AgreementsFilterButton.module.css:2]: np, was hoping we had something like `bg-primary` or what have you.

**Review by xlorepdarkhelm — APPROVED** [review]

**Pull request merged**